crates/wit-component/src/decoding.rs

```diff
@@ -15,7 +15,7 @@ struct ComponentInfo<'a> {
     /// validated.
     types: types::Types,
     /// List of all imports and exports from this component.
-    externs: Vec<(&'a str, Extern<'a>)>,
+    externs: Vec<(ComponentImportName<'a>, Extern<'a>)>,
```

Comment:
Yeah I'm not sure if this was the right way to go initially, but the current thinking is that while the two are the same we'll eventually grow an "implementation" variant of the import name production. That being said it may be better so not try to predict the future and instead only implement what we have today which only one name.